vllm/platforms/cuda.py

```diff
@@ -50,6 +51,65 @@ def wrapper(*args: _P.args, **kwargs: _P.kwargs) -> _R:
     return wrapper
 
 
+def set_cuda_context(device: Union[torch.device, int]) -> bool:
```

Comment:
i would suggest just creating a tensor on the target device, and then call `torch.cuda.set_device` . creating a context this way might interfere with other functionality, e.g. `cuCtxCreate_v2` might be deprecated in the future (in favor of `cuCtxCreate_v3`), and calling the function would break other things easily.